From 47bde186856f3592ce71ba17db527a678477ed62 Mon Sep 17 00:00:00 2001
From: Simon Richter <Simon.Richter@hogyros.de>
Date: Sat, 28 Sep 2013 06:32:24 +0200
Subject: [PATCH 3/3] ICD dispatch table must be first
To: beignet@lists.freedesktop.org

The ICD loader expects the first member of any dispatchable object to be
the dispatch table.
---
 src/cl_mem.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: beignet-0.2+git20130928+187c17e/src/cl_mem.h
===================================================================
--- beignet-0.2+git20130928+187c17e.orig/src/cl_mem.h	2013-09-26 10:55:59.000000000 +0200
+++ beignet-0.2+git20130928+187c17e/src/cl_mem.h	2013-09-28 06:34:22.884556905 +0200
@@ -73,8 +73,8 @@
 #define IS_GL_IMAGE(mem) (mem->type == CL_MEM_GL_IMAGE_TYPE)
 
 typedef  struct _cl_mem {
-  uint64_t magic;           /* To identify it as a memory object */
   DEFINE_ICD(dispatch)
+  uint64_t magic;           /* To identify it as a memory object */
   cl_mem prev, next;        /* We chain the memory buffers together */
   enum cl_mem_type type;
   volatile int ref_n;       /* This object is reference counted */
